### DISCUSSION TITLE
Daylight Saving - cloud history data may be one hour off on day clocks change

### DISCUSSION THREAD

**Discussion by youzer-name** [jasonacox/Powerwall-Dashboard#447] [General]
I think this may have been discussed here last year when the clocks changed, but I couldn't find that discussion so I'm posting a new thread.

TLDR: If you run the tesla-history tool (which I think is also used for cloud mode) your data may be off by 1 hour today.

I run both the normal local PWD stack, getting data from my Gateway 2, and the tesla-history tool, which allows me to get data from my Tesla/Solar City array. I also have a SolarEdge array and I get the data for that from their API using a Python script. My PW gateway sees the total production of both arrays. 

I created a Grafana panel that graphs the two solar sources (stacked) and the solar production reported by the PW gateway. Normally, it should look like this:

![image](https://github.com/jasonacox/Powerwall-Dashboard/assets/57046429/97f97362-7f75-423d-8b0a-0cf51f7f3007)

That was yesterday morning. Everything lines up nicely with the lower-resolution cloud data mostly lining up with the higher-resolution local data, although it is always spikier since it only updates every 15 minutes.

However this morning it looks like:

![image](https://github.com/jasonacox/Powerwall-Dashboard/assets/57046429/552d7902-5483-4907-b2ca-db870dcf0a3c)

The 'Solar City' line, which comes from the tesla-history tool, is one hour off. You can see it showing production from that system before the sun rose (the dotted gray line from the Sun/Moon data source).

I checked the Tesla app on my phone, and sure enough, it shows the solar production starting today at 0700.

![image](https://github.com/jasonacox/Powerwall-Dashboard/assets/57046429/d4f481a3-e871-42ac-abef-2f804485fd9c)

My recollection from last year is that this will most likely resolve itself tomorrow. 

Here in the chart from March 2023 when the clocks changed. This is showing Saturday, Sunday, and Monday. (I didn't get the SolarEdge array until last June, so this chart only has Tesla Cloud and local gateway data)

![image](https://github.com/jasonacox/Powerwall-Dashboard/assets/57046429/00d8bc86-6dca-4704-9aab-9fe4915cf046)

Everything is aligned on Saturday, one hour off on Sunday, and back to normal on Monday.

**Comment by jasonacox** [comment]
Great observation! Thanks @youzer-name .
